lasagne/layers/dnn.py

```diff
@@ -24,7 +24,49 @@ class DNNLayer(Layer):
 
 
 class Pool2DDNNLayer(DNNLayer):
+    """
+    2D pooling layer
 
+    Performs 2D mean- or max-pooling over the two trailing axes of a 4D input
+    tensor. This is an alternative implementation which uses
+    ``theano.sandbox.cuda.dnn.dnn_pool`` directly.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 4D tensor.
```

Comment:
Minor comment: I find "The output if this layer" a bit confusing. As the previous sentence allows both a layer or a shape tuple, does "this layer" refer to the `Pool2DDNNLayer` that is instantiated, maybe? I guess I'd just drop this, as the layer description above already mentions the 4D input tensor.